File: trunk/src/java/org/codehaus/testdox/intellij/actions/BaseToggleAction.java
package org.codehaus.testdox.intellij.actions;

import javax.swing.Icon;

import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.ToggleAction;

abstract class BaseToggleAction extends ToggleAction {

    protected final ActionEvents actionEvents = ActionEvents.instance;

    private final boolean useFromTestDoxToolWindow;

    BaseToggleAction(String text, String description, Icon icon, boolean useFromTestDoxToolWindow) {
        super(text, description, icon);
        this.useFromTestDoxToolWindow = useFromTestDoxToolWindow;
    }

    public void update(AnActionEvent event) {
        super.update(event);

        if (useFromTestDoxToolWindow) {
            actionEvents.getTestDoxToolWindowUI(event).update(event.getPresentation());
        } else {
            actionEvents.getTestDoxController(event).update(event.getPresentation());
        }
    }
}

File: src/scala/org/codehaus/testdox/intellij/actions/BaseToggleAction.scala
package org.codehaus.testdox.intellij.actions

import javax.swing.Icon

import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.ToggleAction

abstract class BaseToggleAction(text: String, description: String, icon: Icon, useFromTestDoxToolWindow: Boolean)
    extends ToggleAction(text, description, icon) {

  protected val actionEvents = ActionEvents.instance

  override def update(event: AnActionEvent) {
    super.update(event)
    selectController(event).update(event.getPresentation())
  }

  private def selectController(event: AnActionEvent) = {
    if (useFromTestDoxToolWindow)
      actionEvents.getTestDoxToolWindowUI(event)
    else
      actionEvents.getTestDoxController(event)
  }
}
